common/SC_ServerBootDelayWarning.cpp

```diff
@@ -0,0 +1,47 @@
+// Add a warning when Windows Defender delays scsynth boot by 60+ seconds
+// cf. github issue #4368
+#include "SC_ServerBootDelayWarning.h"
+
+#ifdef _WIN32
+#    include <iostream>
+#    include <thread>
+#    include <mutex>
+#    include <chrono>
+#    include <condition_variable>
+
+static bool g_isServerBooted = false;
+static std::mutex g_scsynthBootWarningMutex;
+static std::condition_variable g_scsynthBootWarningConditionVariable;
+static std::thread* g_bootWarningThread;
+
+static void displayWarningIfServerHasNotBooted() {
+    std::unique_lock<std::mutex> lock(g_scsynthBootWarningMutex);
+
+    if (!g_scsynthBootWarningConditionVariable.wait_for(lock, std::chrono::seconds(10),
+                                                        [] { return g_isServerBooted; })) {
+        std::cout << "Server: possible boot delay." << std::endl;
+        std::cout << "On some Windows-based machines, Windows Defender sometimes ";
+        std::cout << "delays server boot by one minute." << std::endl;
+        std::cout << "You can add scsynth.exe and/or supernova.exe *processes* to ";
+        std::cout << "Windows Defender exclusion list to disable this check. It's safe." << std::endl;
```

Comment:
consider changing `disable this check` to `avoid this delay` -- it's unclear what `this check` refers to, and because of that, not entirely clear that disabling it means that the delay will go away.

Reply:
good idea !